pycbc/distributions/change_of_variables.py

```diff
@@ -0,0 +1,168 @@
+a change of variables.
+"""
+
+import numpy
+from pycbc import transforms
+from pycbc.distributions import bounded
+
+class ChangeOfVariables(bounded.BoundedDist):
+    """ Class for sampling in one set of parameters and using a Jacobian
+    transform to another set of parameters to calculate the
+    probability density function.
+
+    Examples
+    --------
+    How to initialize a chirp mass and mass ratio to component mass instance:
+
+    >>> from pycbc import transforms
+    >>> from pycbc import distributions
+    >>> mchirp_q_dist = distributions.Uniform(mchirp=(7, 40), q=(1, 10))
+    >>> mass1_mass2_dist = distributions.Uniform(mass1=(8, 46), mass2=(2, 46))
+    >>> cov_dist = distributions.ChangeOfVariables(mchirp_q_dist, mass1_mass2_dist, transforms.MchirpQToMass1Mass2)
+    """
+    name = "change_of_variables"
+
+    def __init__(self, sampling_dist, prior_dist, transform):
+        self._sampling_dist = sampling_dist
+        self._prior_dist = prior_dist
+        self._transform = transform
+        self._bounds = self._sampling_dist._bounds
+        self._params = self._sampling_dist._params
+
+    def _pdf(self, **kwargs):
+        """Returns the pdf at the given values. The keyword arguments must
+        contain all of parameters in self's params. Unrecognized arguments are
+        ignored. Any boundary conditions are applied to the values before the
+        pdf is evaluated.
+        """
+        prior_dist_params = self._transform.convert(kwargs)
+        return self._transform.jacobian(kwargs) \
+                               * self._prior_dist.pdf(**prior_dist_params)
+
+    def _logpdf(self, **kwargs):
+        """Returns the log of the pdf at the given values. The keyword
+        arguments must contain all of parameters in self's params.
+        Unrecognized arguments are ignored. Any boundary conditions are
+        applied to the values before the pdf is evaluated.
+        """
+        prior_dist_params = self._transform.convert(kwargs)
+        return self._transform.jacobian(kwargs) \
+                               + self._prior_dist.logpdf(**prior_dist_params)
+
+    def rvs(self, size=1, param=None):
+        """Gives a set of random values drawn from this distribution.
+
+        Parameters
+        ----------
+        size : {1, int}
+            The number of values to generate; default is 1.
+        param : {None, string}
+            If provided, will just return values for the given parameter.
+            Otherwise, returns random values for each parameter.
+
+        Returns
+        -------
+        structured array
+            The random values in a numpy structured array. If a param was
+            specified, the array will only have an element corresponding to the
+            given parameter. Otherwise, the array will have an element for each
+            parameter in self's params.
+        """
+        return self._sampling_dist.rvs(size=size, param=param)
```

Comment:
Keeping with my comment above, this should generate the rvs values from the `prior_dist.rvs`, then apply the transform to get the prior in the new distribution.